### PR TITLE
Update loads and store immediates to be alignment varint followed by …

### PR DESCRIPTION
**Pull request by titzer**
…index varint.

Relies on https://codereview.chromium.org/1775123003/

### CODE REVIEW

File: test/spec/select.txt [hunk 1]
```diff
@@ -3,9 +3,12 @@
 ;;; FLAGS: --spec
 ;;; STDIN_FILE: third_party/testsuite/select.wast
 (;; STDERR ;;;
-
+Failed to open startup resource '/usr/local/google/home/titzer/v8/wasm/sexpr-wasm-prototype/out/snapshot_blob.bin'.
```
Comment (binji) [inline test/spec/select.txt]: hm...

File: src/wasm-binary-writer.c [hunk 2]
```diff
@@ -701,23 +701,15 @@ static void write_expr(WasmWriteContext* ctx,
...
     case WASM_EXPR_TYPE_LOAD: {
-      /* Access byte: 0bAaao0000
-       A = Alignment. If set, access is unaligned
-       a = Atomicity. 0 = None, 1 = SeqCst, 2 = Acq, 3 = Rel
-       o = Offset. If set, offset field follows access byte */
       out_opcode(ws, s_mem_opcodes[expr->load.op.op_type]);
-      uint32_t natural_align = expr->load.op.size >> 3;
       uint32_t align = expr->load.align;
-      uint8_t access = 0;
-      if (align == WASM_USE_NATURAL_ALIGNMENT)
```
Comment (binji) [inline src/wasm-binary-writer.c]: you'll still need to keep this if

Comment (titzer) [inline src/wasm-binary-writer.c]: Done. PTAL

File: src/wasm-binary-writer.c [hunk 3]
```diff
@@ -757,20 +752,18 @@ static void write_expr(WasmWriteContext* ctx,
...
-      if (align < natural_align)
-        access |= 0x80;
-      if (expr->store.offset)
-        access |= 0x10;
-      out_u8(ws, access, "store access byte");
-      if (expr->store.offset)
-        out_u32_leb128(ws, (uint32_t)expr->store.offset, "store offset");
+      uint32_t align = expr->store.align;
+      if (align == WASM_USE_NATURAL_ALIGNMENT) {
+        align = expr->load.op.size >> 3;
```
Comment (binji) [inline src/wasm-binary-writer.c]: should be store

File: src/wasm-binary-writer.c [hunk 4]
```diff
@@ -757,20 +752,18 @@ static void write_expr(WasmWriteContext* ctx,
...
+      }
+      uint8_t align_log = 0;
+      while (align) {
+        align >>= 1;
+        align_log++;
+      }
+      out_u8(ws, align_log, "alignment");
+      out_u32_leb128(ws, (uint32_t)expr->load.offset, "store offset");
```
Comment (binji) [inline src/wasm-binary-writer.c]: here too

File: src/wasm-binary-writer.c (hunk 3 above)
Comment (titzer) [inline src/wasm-binary-writer.c]: Done.

File: src/wasm-binary-writer.c (hunk 4 above)
Comment (titzer) [inline src/wasm-binary-writer.c]: Done.

File: test/dump/basic.txt [hunk 5]
```diff
@@ -27,34 +27,36 @@
...
 000001a: 33                                         ; OPCODE_I32_STORE_MEM
-000001b: 00                                         ; store access byte
-000001c: 0a                                         ; OPCODE_I32_CONST
-000001d: 00                                         ; i32 literal
-000001e: 40                                         ; OPCODE_I32_ADD
-000001f: 2a                                         ; OPCODE_I32_LOAD_MEM
-0000020: 00                                         ; load access byte
-0000021: 0a                                         ; OPCODE_I32_CONST
-0000022: 00                                         ; i32 literal
+000001b: 03                                         ; alignment
```
Comment (binji) [inline test/dump/basic.txt]: did you re-run the tests? Shouldn't this be 2?

**Comment by binji** [comment]
lgtm w/ note

File: test/dump/basic.txt (hunk 5 above)
Comment (titzer) [inline test/dump/basic.txt]: Just blindly -r'd. Twers off by one methinks.

**Comment by binji** [comment]
Cool, thanks.

**Pull request merged**